### PR TITLE
New enforcer sprites, adds suppressor and flashlight support, and a trainer

### PR DESCRIPTION
**Pull request by shazbot194**
sagbg16nsj69bt4bm1dh1a

This PR adds in some new Enforcer sprites that are meant to be in line with our current pistols. It also adds in some "unique_skins" with several colors you get the same way you would get them from the detective revolver, by alt clicking on the gun, however you cannot name it, since its not truely a unique gun.

Just as a side note, I can add more colors before this is merged so if you really want a color thats not too far out there, tell me and I can add it.

Another side note, if people want it and maintainers are ok with it, I can make it suppresserable.

_kolenbdq1gkih4puyc8xg

Update:7/25/2017

So I have taken the time to add some more features, slightly movining this PR away from just updating the sprites, and I wil be changing the title to reflect this. First of all, the smallest part of this update is there are now three new colors. These are accessed just like any other color.
lkqobxr0rnshvl0rjhfdcq

Second, I have added the ability to suppress Enforcers. This works with all colors and can be turned off by changing the value from TRUE to FALSE. If it is officially requested, I can turn it off before merging.
4cwhfpksrnupgtnjyrcxsw

Last but by no means least, I have added a new version of the Enforcer. This blue training version doesn't load .45, but rather loads foam darts. This means it's not only a 100% non-lethal means of subdueing people, but is can be used as a trainer, for when someone can't be trusted with rubber bullets, is too green and might hurt themselves with a real firearm, or when ammo is scarce. It does come pre-loaded with riot darts.
z21ocs7wsvoaalqmw4h8oq

Edit: 7/26/2017

Now you can go full tacti-cool and add a flashlight to your Enforcer. It works just like how it does on any las gun. No longer does having a a light attachment point factor into your choice if you want a tazer or Enforcer. The trainer can also have a flashlight attached.

Small update:
The flashlight sprite is now attached to the gun rather then being suspended in addtion to removing the light particles, and the riot bullets in the magazine are closer to their real color rather than being bright red.
kkkf9owdtfa2eqtfc2sr6q
xsyh3eedrv6kg3e8rdqg5w

:cl:  Shazbot
Change: New enforcer sprites
Change: Enforcers can now be suppressed
Change: Adds in a trainer version of the Enforcer
/:cl:

### CODE REVIEW

**Comment by pinatacolada** [comment]
we cs:go now

**Comment by Ralta** [comment]
Glocktastic.
Glocktacular.
Glockerrific.
Glocktabulous.
Glockazing.
Glockedible.
Glockarkable.
btw I think these looks like Glocks. Cool shit.

**Comment by Xantholne** [comment]
@shazbot194 I believe some people actually wanted it suppressorable so I'd probably do that after poking a few of the maints

**Comment by taukausanake** [comment]
PUT FLAMES ON ONE OF THEM. Everything looks cooler with flames! :D

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 1]
```diff
@@ -32,6 +32,29 @@
 	magazine = new /obj/item/ammo_box/magazine/m10mm/empty(src)
 	..()
 
+/obj/item/weapon/gun/projectile/automatic/pistol/enforcer
+	name = "Enforcer .45"
+	desc = "A pistol of modern design."
+	icon_state = "enforcer_grey"
+	force = 10
+	mag_type = /obj/item/ammo_box/magazine/m45/enforcer45
+	can_suppress = 0
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/projectile/pistol.dm]: `FALSE`

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 2]
```diff
@@ -32,6 +32,29 @@
...
 	..()
 
+/obj/item/weapon/gun/projectile/automatic/pistol/enforcer
+	name = "Enforcer .45"
+	desc = "A pistol of modern design."
+	icon_state = "enforcer_grey"
+	force = 10
+	mag_type = /obj/item/ammo_box/magazine/m45/enforcer45
+	can_suppress = 0
+	unique_reskin = 1
```
Comment (Fox-McCloud) [inline code/modules/projectiles/guns/projectile/pistol.dm]: `TRUE`

**Comment by ktoma36** [comment]
I think it would be cool if you could attach a sec light to it somewhere, probably under the barrel. Just some extra functionality :). Other than that it looks great!

**Comment by Anasari** [comment]
Attachable seclite would be amazing indeed and make it possible to replace sec gun in a special event

**Comment by shazbot194** [comment]
Well, tomorrow, I'll have a look to see if it won't be too hard to add sec light attach-ability, but no promises since there is currently no system for ballistic weapons to do that.

**Comment by FlattestGuitar** [comment]
I'm iffy on the colors. Everything except blue looks really washed out and weird.

Also, the riot darts are a straight upgrade over rubber bullets that thing takes. Should probably give people a reason to want to use rubber bullets instead of riot darts.

**Comment by ktoma36** [comment]
Rubber bullets are almost always a two hit stun, riot darts can take 4-5

**Comment by shazbot194** [comment]
Well, the real ones can still load lethal ammo, something the training one cannot do at all, not to mention you can get a very similar effect by getting a normal dart pistol and loading it with riot darts. That being said, we could always buff up the stamina damage a little bit on the rubber bullets as well and make the rubber .45 do stamina damage closer to the rethal .45.

**Comment by Xantholne** [comment]
IMO the flashlight attachement needs moved one moe sprite up but everything else is fine.

**Comment by FalseIncarnate** [comment]
Training version is missing the orange barrel tip to help reinforce it isn't a real gun. (Airsoft gun logic)

**Comment by shazbot194** [comment]
Well, the current foam pistol uses the same grey colored barrel, I could attempt to recolor it, but I think it might be better to keep it kinda consistent between the foam pistols.

**Comment by FalseIncarnate** [comment]
Was mostly a joke, looks goopd

**Comment by taukausanake** [comment]
I was going to mention I've NEVER seen orange tipped training guns before. I think them being blue is enough

**Comment by Allfd** [comment]
I just want to say, ❤️

**Comment by shazbot194** [comment]
So, any updates on this Pull? I haven't heard anything about it since Allfd's comment.

**Comment by Spacemanspark** [comment]
@Takausanake The blank firing adapters we attach to our rifles when we go to do non-live fire training are orange, actually.

**Pull request merged**